tasks/sealsupra/task_supraseal.go

```diff
@@ -0,0 +1,535 @@
+		}
+
+		if len(ownedBy) != 1 {
+			return false, xerrors.Errorf("no machine found for task %d", taskID)
+		}
+
+		for i, sector := range sectors {
+			var commr [32]byte
+			if !supraffi.GetCommR(commr[:], outPaths[i].Cache) {
+				return false, xerrors.Errorf("getting commr failed")
+			}
+
+			sealedCID, err := commcid.ReplicaCommitmentV1ToCID(commr[:])
+			if err != nil {
+				return false, xerrors.Errorf("getting sealed CID: %w", err)
+			}
+
+			_, err = tx.Exec(`UPDATE sectors_sdr_pipeline SET after_sdr = TRUE, after_tree_c = TRUE, after_tree_r = TRUE, after_tree_d = TRUE,
+                                ticket_epoch = $3, ticket_value = $4, tree_d_cid = $5, tree_r_cid = $6, task_id_sdr = NULL, task_id_tree_r = NULL, task_id_tree_c = NULL, task_id_tree_d = NULL
+                            WHERE sp_id = $1 AND sector_number = $2`, sector.SpID, sector.SectorNumber, ticketEpochs[i], tickets[i], unsealedCID.String(), sealedCID)
+			if err != nil {
+				return false, xerrors.Errorf("updating sector: %w", err)
+			}
+
+			// insert batch refs
+			_, err = tx.Exec(`INSERT INTO batch_sector_refs (sp_id, sector_number, machine_host_and_port, pipeline_slot)
+    						  VALUES ($1, $2, $3, $4) ON CONFLICT DO NOTHING`, sector.SpID, sector.SectorNumber, ownedBy[0].HostAndPort, slot)
+			if err != nil {
+				return false, xerrors.Errorf("inserting batch refs: %w", err)
+			}
+		}
+
+		return true, nil
+	}, harmonydb.OptionRetry())
+	if err != nil {
+		return false, xerrors.Errorf("persisting success: %w", err)
+	}
+
+	cleanup = func() {
+		s.outSDR.Unlock()
+		// NOTE: We're not releasing the slot yet, we keep it until sector Finalize
+	}
+
+	return true, nil
+}
+
+func (s *SupraSeal) CanAccept(ids []harmonytask.TaskID, engine *harmonytask.TaskEngine) (*harmonytask.TaskID, error) {
+	if s.slots.Available() == 0 {
+		return nil, nil
+	}
+
+	// check if we have enough huge pages available
+	// sysctl vm.nr_hugepages should be >= 36 for 32G sectors
+	if err := hugepageutil.CheckHugePages(36); err != nil {
+		log.Warnw("huge pages check failed, try 'sudo sysctl -w vm.nr_hugepages=36' and make sure your system uses 1G huge pages", "err", err)
+		return nil, nil
+	}
+
+	id := ids[0]
+	return &id, nil
+}
+
+var ssizeToName = map[abi.SectorSize]string{
+	abi.SectorSize(2 << 10):   "2K",
+	abi.SectorSize(8 << 20):   "8M",
+	abi.SectorSize(512 << 20): "512M",
+	abi.SectorSize(32 << 30):  "32G",
+	abi.SectorSize(64 << 30):  "64G",
+}
+
+func (s *SupraSeal) TypeDetails() harmonytask.TaskTypeDetails {
+	return harmonytask.TaskTypeDetails{
+		Max:  s.pipelines,
+		Name: fmt.Sprintf("Batch%d-%s", s.sectors, ssizeToName[must.One(s.spt.SectorSize())]),
+		Cost: resources.Resources{
+			Cpu: 1,
+			Gpu: 0,
+			Ram: 16 << 30,
+		},
+		MaxFailures: 4,
+		IAmBored:    passcall.Every(30*time.Second, s.schedule),
+	}
+}
+
+func (s *SupraSeal) Adder(taskFunc harmonytask.AddTaskFunc) {
+}
+
+func (s *SupraSeal) schedule(taskFunc harmonytask.AddTaskFunc) error {
```

Comment:
when does this get called? what's its contract?

Reply:
> IAmBored:    passcall.Every(30*time.Second, s.schedule),